### PR TITLE
docs: add `deferred`, `range`, `e`, and `pi` to API reference

### PR DESCRIPTION
**Pull request by tswast**
## Description of changes

Some top-level APIs were missing from the API reference. This adds them.

### CODE REVIEW

File: docs/presentations-listing.json
Comment (cpcloud) [inline docs/presentations-listing.json:1]: Is this file needed/related to the PR's motivation?

Comment (tswast) [reply docs/presentations-listing.json:1]: No, `just docs-preview` seems to have created that. I can delete.

**Comment by tswast** [comment]
Re: `file:///home/runner/work/ibis/ibis/docs/_output/reference/range | Failed: Cannot find file`. I'll try some fixes. Adding a more specific package name seemed to help for the others.

**Comment by cpcloud** [comment]
That means that there's a file referenced somewhere that doesn't exist. That usually means that wherever you're referencing `range`, the way you've referenced it is incorrect.

**Comment by cpcloud** [comment]
Ah, looks like you're not referencing `range` anywhere, so that may not be the issue.

**Comment by gforsyth** [comment]
Looks like the docstring for `ibis.range` has an invalid link -- let me poke it

**Comment by gforsyth** [comment]
I was wrong about interlinks, we do have them, so it may just be how we're spelling it.  Poking locally, still

**Comment by tswast** [comment]
Thanks @gforsyth I see you're updating the PR.

Looks like https://docs.python.org/3/library/functions.html#func-range would be the right link, but yeah if interlinks work it would be good to reference the built-in that way.

**Comment by gforsyth** [comment]
ok, that took way too long to figure out, but it's rendering correctly for me now

File: ibis/expr/api.py [hunk 1]
```diff
@@ -2094,8 +2094,8 @@ def range(start, stop, step) -> ir.ArrayValue:
     Integer ranges are supported, as well as timestamp ranges.
 
     ::: {.callout-note}
-    `start` is inclucive and `stop` is exclusive, just like Python's builtin
-    [`range`](range).
+    `start` is inclusive and `stop` is exclusive, just like Python's builtin
+    [`range`](:py:class:`range`).
```
Comment (cpcloud) [inline ibis/expr/api.py:2098]: The original is correct here. You can't use Sphinx syntax in quartodoc.

Comment (cpcloud) [inline ibis/expr/api.py:2098]:
```suggestion
    [](`range`).
```

I believe this should fix the reference error.

Comment (cpcloud) [reply ibis/expr/api.py:2098]: Ah, didn't see your comment in the top-level PR.

I think we should probably avoid the Sphinx syntax since it kind of gives the impression that people should write Sphinx.

Admittedly, it's probably unlikely that someone will come across this and then additionally conclude that Sphinx is in use, but if there's a way we can do it that matches everything else like it (referencing a builtin object) then I think we should.

Comment (gforsyth) [reply ibis/expr/api.py:2098]: I swear I tried the suggested code before and it didn't render, but I went back to confirm and it works.

Interlinks on quartodoc do support the sphinx syntax, just in case we ever need it for specifying an ambiguous link, but I'll switch this one to use the simpler syntax.

Comment (cpcloud) [reply ibis/expr/api.py:2098]: > Interlinks on quartodoc do support the sphinx syntax

😮

**Review by cpcloud — APPROVED** [review]
